app/controllers/ApplicationController.scala

```diff
@@ -274,6 +274,14 @@ trait ApplicationController extends BaseController with GatekeeperAuthWrapper {
     applicationService.fetchApplication(appId).flatMap(f)
   }
 
+  private def withRestrictedApp(appId: String)(f: ApplicationWithHistory => Future[Result])(implicit request: Request[_]) = {
+    applicationService.fetchApplication(appId).flatMap { app => app.application.access match {
+      case _: Standard => f(app)
+      case _ if isSuperUser => f(app)
+      case _ => Future.successful(Unauthorized(views.html.unauthorized()))
+    }}
+  }
+
```

Comment:
withRestrictedApp seems like a specialisation of withApp and therefore may be written in terms of withApp

```
  private def withRestrictedApp(appId: String)(f: ApplicationWithHistory => Future[Result])(implicit request: Request[_]) =
    withApp(appId) { app => app.application.access match {
      case _: Standard => f(app)
      case _ if isSuperUser => f(app)
      case _ => Future.successful(Unauthorized(views.html.unauthorized()))
    }
  }
```

Maybe?